File: backend/test-number-path.js
/**
 * 测试 Number Path API
 */

const url = 'http://localhost:3000/api/worksheets/generate-image';

const payload = {
    categoryId: 'math',
    pageTypeId: 'number-path',
    config: {
        theme: 'dinosaur',
        maxNumber: 20,
        pageCount: 1
    }
};

async function test() {
    console.log('Testing Number Path API...');
    console.log('URL:', url);
    console.log('Payload:', JSON.stringify(payload, null, 2));
    
    try {
        const response = await fetch(url, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify(payload)
        });
        
        console.log('\nStatus:', response.status);
        const data = await response.json();
        console.log('Response:', JSON.stringify(data, null, 2));
    } catch (error) {
        console.error('Error:', error.message);
    }
}

test();

File: scripts/test_number_path.py
"""测试 Number Path API"""
import requests
import json

url = "http://localhost:3000/api/worksheets/generate-image"

payload = {
    "categoryId": "math",
    "pageTypeId": "number-path",
    "config": {
        "theme": "dinosaur",
        "maxNumber": 20,
        "pageCount": 1
    }
}

print("Testing Number Path API...")
print(f"URL: {url}")
print(f"Payload: {json.dumps(payload, indent=2)}")

try:
    response = requests.post(url, json=payload, timeout=120)
    print(f"\nStatus: {response.status_code}")
    data = response.json()
    print(f"Response: {json.dumps(data, indent=2, ensure_ascii=False)}")
except Exception as e:
    print(f"Error: {e}")
